src/beast/beast/hash/hash_append.h

```diff
@@ -21,125 +21,165 @@
-#endif
 #include <array>
 #include <cstdint>
+#include <cstring>
 #include <functional>
+#include <map>
 #include <memory>
+#include <set>
 #include <string>
+#include <system_error>
 #include <tuple>
+#include <unordered_map>
+#include <unordered_set>
 #include <beast/cxx14/type_traits.h> // <type_traits>
 #include <beast/cxx14/utility.h> // <utility>
 #include <vector>
 
-// Set to 1 to disable variadic hash_append for tuple. When set, overloads
-// will be manually provided for tuples up to 10-arity. This also causes
-// is_contiguously_hashable<> to always return false for tuples.
-//
-#ifndef BEAST_NO_TUPLE_VARIADICS
-# ifdef _MSC_VER
-#  define BEAST_NO_TUPLE_VARIADICS 1
-#  ifndef BEAST_VARIADIC_MAX
-#   ifdef _VARIADIC_MAX
-#    define BEAST_VARIADIC_MAX _VARIADIC_MAX
-#   else
-#    define BEAST_VARIADIC_MAX 10
-#   endif
-#  endif
-# else
-#  define BEAST_NO_TUPLE_VARIADICS 0
-# endif
-#endif
-
-// Set to 1 if std::pair fails the trait test on a platform.
-#ifndef BEAST_NO_IS_CONTIGUOUS_HASHABLE_PAIR
-#define BEAST_NO_IS_CONTIGUOUS_HASHABLE_PAIR 0
-#endif
+namespace beast {
 
-// Set to 1 if std::tuple fails the trait test on a platform.
-#ifndef BEAST_NO_IS_CONTIGUOUS_HASHABLE_TUPLE
-# ifdef _MSC_VER
-#  define BEAST_NO_IS_CONTIGUOUS_HASHABLE_TUPLE 1
-# else
-#  define BEAST_NO_IS_CONTIGUOUS_HASHABLE_TUPLE 0
-# endif
-#endif
+namespace detail {
 
-namespace beast {
+template <class T>
+/*constexpr*/
+inline
+void
+reverse_bytes(T& t)
+{
+    unsigned char* bytes = static_cast<unsigned char*>(std::memmove(std::addressof(t),
+                                                                    std::addressof(t),
+                                                                    sizeof(T)));
+    for (unsigned i = 0; i < sizeof(T)/2; ++i)
+        std::swap(bytes[i], bytes[sizeof(T)-1-i]);
+}
 
-/** Metafunction returning `true` if the type can be hashed in one call.
+template <class T>
+/*constexpr*/
+inline
+void
+maybe_reverse_bytes(T& t, std::true_type)
+{
+}
 
-    For `is_contiguously_hashable<T>::value` to be true, then for every
-    combination of possible values of `T` held in `x` and `y`,
-    if `x == y`, then it must be true that `memcmp(&x, &y, sizeof(T))`
-    return 0; i.e. that `x` and `y` are represented by the same bit pattern.
-   
-    For example:  A two's complement `int` should be contiguously hashable.
-    Every bit pattern produces a unique value that does not compare equal to
-    any other bit pattern's value.  A IEEE floating point should not be
-    contiguously hashable because -0. and 0. have different bit patterns,
-    though they compare equal.
-*/
-/** @{ */
-// scalars
 template <class T>
-struct is_contiguously_hashable
-    : public std::integral_constant <bool,
-        std::is_integral<T>::value || 
-        std::is_enum<T>::value     ||
-        std::is_pointer<T>::value>
+/*constexpr*/
+inline
+void
+maybe_reverse_bytes(T& t, std::false_type)
 {
-};
+    reverse_bytes(t);
+}
```

Comment:
Since we're converting to an `integral_constant`, `::value` isn't needed.  Had we been converting to a `bool`, then `::value` *would* have been needed.